### ISSUE TITLE
pre-training  data of Unixcoder

### ISSUE THREAD

**Issue by skye95git** [microsoft/CodeBERT#134]
Do you have plan to release pre-training data, particularly `ﬂattened AST`? 

Are there plans to publish the code for `AST Mapping Function F`?

**Comment by guoday** [comment]
[code.zip](https://github.com/microsoft/CodeBERT/files/8700184/code.zip). Follow the Readme to download pre-training data. And then you can use `preprocess` function in `preprocess.py` to get flattened AST.

**Comment by skye95git** [comment]
> [code.zip](https://github.com/microsoft/CodeBERT/files/8700184/code.zip). Follow the Readme to download pre-training data. And then you can use `preprocess` function in `preprocess.py` to get flattened AST.

Thanks for your reply! I have some other questions:
1. Is the tokenizer used for `def preprocess(js,tokenizer,lang)` the same tokenizer used for pre-training? such as `We follow Liu et al. (2019) to train a byte-pair encoding vocabulary (Sennrich et al., 2015) with 50K subword units for programming languages and add 1,416 additional special tokens into the vocabulary to represent non-terminal symbols in AST.`


2. > [unixcoder-base-nine](https://huggingface.co/microsoft/unixcoder-base-nine): Continue pre-training unixcoder-base-unimodal on NL-PL pairs of CodeSearchNet dataset and additional 1.5M NL-PL pairs of C, C++ and C# programming language. The model can support nine languages: java, ruby, python, php, javascript, go, c, c++ and c#.   

How to download `1.5M NL-PL pairs of C, C++ and C# `?

**Comment by guoday** [comment]
1. Yes
2. The datasets are collected from GitHub. Since of license issues, we can't release these data.

**Comment by skye95git** [comment]
Was Unixcoder pre-train from scratch, or was it initialized with Graphcodebert?

**Comment by skye95git** [comment]
`add 1,416 additional special tokens into the vocabulary to represent non-terminal symbols in AST`
Are the 1,416 special tokens manually selected by you? Are they manually added to the vocabulary?

**Comment by guoday** [comment]
Use the tree-sitter tool to parse CodeSearchNet corpus into AST and get all internal nodes of AST.  Finally, there are 1,416 types of nodes of AST.

**Comment by skye95git** [comment]
> Use the tree-sitter tool to parse CodeSearchNet corpus into AST and get all internal nodes of AST. Finally, there are 1,416 types of nodes of AST.

Thanks for your reply! Will the merges.txt file need to be changed when 1,416 special tokens are added to vocabulary?

**Comment by skye95git** [comment]
![image](https://user-images.githubusercontent.com/41561936/177481772-79af823c-b273-434c-881d-1eb78000ebb1.png)
1. Is graphCodebert pre-trained to use 2.3M of annotated code directly from Codesearchnet? Do you use 6.4m Unimodal codes in pre-training?    


2. > For fair comparison, we use the same dataset as CodeBERT (Feng et al., 2020) to pretrain our model.   

GraphcodeBert used 2.3m bimodal DATA, but Codebert only used 2.1m Bimodal DATA. Did Codebert clean bimodal DATA before pre-training?

- Dataset Size Statistics in CodeSearchNet:
![image](https://user-images.githubusercontent.com/41561936/177484182-d51bc1e8-3016-4587-9d76-786807c70cc3.png)

   
- Statistics of the dataset in CodeBert:
![image](https://user-images.githubusercontent.com/41561936/177484054-44d75f8c-6952-4693-a84f-1792bfb18503.png)

**Comment by guoday** [comment]
> ![image](https://user-images.githubusercontent.com/41561936/177481772-79af823c-b273-434c-881d-1eb78000ebb1.png)
> 
> 1. Is graphCodebert pre-trained to use 2.3M of annotated code directly from Codesearchnet? Do you use 6.4m Unimodal codes in pre-training?
> 2. > For fair comparison, we use the same dataset as CodeBERT (Feng et al., 2020) to pretrain our model.
> 
> GraphcodeBert used 2.3m bimodal DATA, but Codebert only used 2.1m Bimodal DATA. Did Codebert clean bimodal DATA before pre-training?
> 
> * Dataset Size Statistics in CodeSearchNet:
>   ![image](https://user-images.githubusercontent.com/41561936/177484182-d51bc1e8-3016-4587-9d76-786807c70cc3.png)
> * Statistics of the dataset in CodeBert:
>   ![image](https://user-images.githubusercontent.com/41561936/177484054-44d75f8c-6952-4693-a84f-1792bfb18503.png)

1. CodeBERT and GraphCodeBERT use 2.1M NL-PL pairs. We exclude dev and test set. For GraphCodeBERT, we don't use 6.4M unimodal code.
2. Answered in 1.

**Comment by guoday** [comment]
> ![image](https://user-images.githubusercontent.com/41561936/177484383-89fa063e-03ec-4016-b7b0-47747bcfd50c.png)
> 
> Is the cleaned data only used in the fine-tuning part? Does the GraphcodeBert pre-training data need to be cleaned?

The cleaned data is only used for code summarizaiton task.
